google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/PubSub.java

```diff
@@ -137,6 +140,49 @@ public static PullOption executorFactory(ExecutorFactory executorFactory) {
     }
   }
 
+  @AutoValue
+  public abstract static class FlowControlSettings {
```

Comment:
Why can't this be a GAX class? It could be made generic like BundlingSettings by renaming 'messages' to 'elements'. Then `maxOutstandingBytes` corresponds to `requestByteThreshold` and `maxOutstandingElements` corresponds to `elementCountThreshold`. In fact, a better matchup might be `requestByteMaxOutstanding` and `elementCountMaxOutstanding`.